### PR TITLE
Allow GET requests for apollo server

### PR DESCRIPTION
**Pull request by devotox**

### CODE REVIEW

**Comment by targos** [comment]
what's the goal of this change?

**Comment by devotox** [comment]
> what's the goal of this change?

To allow a user to use a GET request instead of a POST to run a GQL operation.

This is sometimes beneficial as you are able to cache these on proxy servers and on the client

File: src/ApolloServer.ts [hunk 1]
```diff
@@ -133,7 +137,7 @@ export default class ApolloServer extends ApolloServerBase {
   }
 
   public getUploadsMiddleware() {
-    return async (ctx: HttpContextContract, next: () => Promise<void>) => {
+    return async (ctx: HttpContextContract, next: () => void) => {
```
Comment (targos) [inline src/ApolloServer.ts:140]: That change doesn't seem related

File: src/ApolloServer.ts [hunk 2]
```diff
@@ -99,8 +99,12 @@ export default class ApolloServer extends ApolloServerBase {
   }
 
   public applyMiddleware({ Route }: ServerRegistration): void {
-    Route.get(this.$path, this.getPlaygroundHandler());
+    const playgroundPath = `${this.$path}/playground`;
+    Route.get(playgroundPath, this.getPlaygroundHandler());
```
Comment (targos) [inline src/ApolloServer.ts:103]: Have you tested that the playground still works at the new URL?

Comment (devotox) [reply src/ApolloServer.ts:103]: I tested it in my application and there does not seem to be an issue

File: src/ApolloServer.ts (hunk 1 above)
Comment (devotox) [reply src/ApolloServer.ts:140]: I believe there was a lint error

**Review by devotox — COMMENTED** [review]
Something I was able to do as well on the application side was this

```
import ApolloServer from '@ioc:Apollo/Server';

// ApolloServer.applyMiddleware({ Route });

Route.any('graphql', ApolloServer.getGraphqlHandler())
  .middleware(ApolloServer.getUploadsMiddleware());

Route.get('graphql/playground', ApolloServer.getPlaygroundHandler());
```

File: src/ApolloServer.ts (hunk 1 above)
Comment (devotox) [reply src/ApolloServer.ts:140]: Screenshot 2021-05-14 at 11 40 19

**Review by targos — APPROVED** [review]

**Comment by targos** [comment]
Thanks for the pull request! The change is now available in v0.14.0
